cmd/e2e/utilities_test.go

```diff
@@ -269,14 +272,25 @@ func allocate(buildID, sessionID string, cert tls.Certificate) error {
 	transport := &http.Transport{TLSClientConfig: tlsConfig}
 	client := &http.Client{Transport: transport}
 
+	//Block of code to make port grabbing dynamic
```

Comment:
Any chance we can use the API to get the Port and have this code in the utilities_test.go file? Something like this might work:

```golang
svc := corev1.Service{}
	if err := kubeClient.Get(ctx, types.NamespacedName{Name: "...", Namespace: "..."}, &svc); err != nil {
		return err
	}
``` 

Main reasons are that we don't use `kubectl` anywhere else in the code (apart from the e2e shell scripts). Appreciated!

Reply:
> Now that I think about it though, an alternative solution would be NOT to define the port in the Makefile. We can do something like that on the `pkg/operator/config/manager/manager.yaml` file:
> 
> ```yaml
> # rest of YAML file...
>  - name: ALLOC_API_SVC_PORT
>     value: "5000"
>   - name: TLS_SECRET_NAMESPACE
>     valueFrom:
>       fieldRef:
>         fieldPath: metadata.namespace
> name: manager
> securityContext:
>   allowPrivilegeEscalation: false
> livenessProbe:
>   httpGet:
>     path: /healthz
>     port: 8081
>   initialDelaySeconds: 15
>   periodSeconds: 20
> readinessProbe:
>   httpGet:
>     path: /readyz
>     port: 8081
>   initialDelaySeconds: 5
>   periodSeconds: 10
> resources:
>   requests:
>     cpu: 100m
>     memory: 500Mi
>   limits:
>     cpu: 1000m
>     memory: 500Mi
> ports:
>   - containerPort: 5000
>     hostPort: 5000
> # rest of YAML file...
> ```
> 
> People that use the port 5000 can manually change it on this file. Agreed, it's more cumbersome than customizing directly but I think it partially solves the problem. Can we do it this way? Left some more comments as well. Thanks again!

This was my original approach. However, I found that every instance of "5000" in the manager.yaml needs to be replaced. The goal with modifying the makefile was for users to only have to change the port in one spot. A different approach is we could update the documentation to state something along the lines of "To change the port, search and replace every instance of '5000' inside the manager.yaml." I'll continue to adjust the code to make it grab the port from corev1, rather than rely on that env variable.

Reply:
Thanks. Yeah, it's kinda of a mess to change, but better than having an "int" and a "string" variables.